app/ts/utils/solidityTypes.ts

```diff
@@ -114,46 +114,34 @@ function getSolidityTypeCategory(type: SolidityType) {
 
 export async function parseSolidityValueByTypeEnriched(ethereumClientService: EthereumClientService, type: SolidityType, value: unknown, userAddressBook: UserAddressBook, isArray: boolean, useLocalStorage: boolean = true): Promise<EnrichedGroupedSolidityType> {	
 	const categorized = getSolidityTypeCategory(type)
-	if (isArray) {
-		switch (categorized) {
-			case 'address': return { type: `${ categorized }[]`, value: await Promise.all(funtypes.ReadonlyArray(EthereumAddress).parse(value).map((value) => identifyAddress(ethereumClientService, userAddressBook, value, useLocalStorage))) }
-			case 'bool': return { type: `${ categorized }[]`, value: funtypes.ReadonlyArray(NonHexBigInt).parse(value).map((a) => a === 1n) }
-			case 'bytes': return { type: `${ categorized }[]`, value: funtypes.ReadonlyArray(EthereumData).parse(value) }
-			case 'fixedBytes': return { type: `${ categorized }[]`, value: funtypes.ReadonlyArray(EthereumData).parse(value) }
-			case 'integer': return { type: `${ categorized }[]`, value: funtypes.ReadonlyArray(funtypes.Union(NonHexBigInt, funtypes.Number, funtypes.BigInt)).parse(value).map((x) => BigInt(x)) }
-			case 'string': return { type: `${ categorized }[]`, value: funtypes.ReadonlyArray(funtypes.String).parse(value) }
-			default: assertNever(categorized)
-		}
-
-	}
-	switch (categorized) {
-		case 'address': return { type: categorized, value: await identifyAddress(ethereumClientService, userAddressBook, EthereumAddress.parse(value), useLocalStorage) }
-		case 'bool': return { type: categorized, value: NonHexBigInt.parse(value) === 1n }
-		case 'bytes': return { type: categorized, value: EthereumData.parse(value) }
-		case 'fixedBytes': return { type: categorized, value: EthereumData.parse(value) }
-		case 'integer': return { type: categorized, value: BigInt(funtypes.Union(NonHexBigInt, funtypes.Number, funtypes.BigInt).parse(value)) }
-		case 'string': return { type: categorized, value: funtypes.String.parse(value) }
-		default: assertNever(categorized)
+	if (categorized === 'address') {
+		if (isArray) return { type: `${ categorized }[]`, value: await Promise.all(funtypes.ReadonlyArray(EthereumAddress).parse(value).map((value) => identifyAddress(ethereumClientService, userAddressBook, value, useLocalStorage))) }
+		return { type: categorized, value: await identifyAddress(ethereumClientService, userAddressBook, EthereumAddress.parse(value), useLocalStorage) }
```

Comment:
within the "if" condition, value for `categorized` will always be `'address'` anyway
```suggestion
	if (categorized === 'address') {
		if (isArray) return { type: `address[]`, value: await Promise.all(funtypes.ReadonlyArray(EthereumAddress).parse(value).map((value) => identifyAddress(ethereumClientService, userAddressBook, value, useLocalStorage))) }
		return { type: 'address', value: await identifyAddress(ethereumClientService, userAddressBook, EthereumAddress.parse(value), useLocalStorage) }
```